pkg/tsdb/cloudwatch/routes/dimension_keys.go

```diff
@@ -0,0 +1,66 @@
+package routes
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/services"
+)
+
+func DimensionKeysHandler(rw http.ResponseWriter, req *http.Request, clientFactory models.ClientsFactoryFunc, pluginCtx backend.PluginContext) {
```

Comment:
I would suggest declaring a DimensionKeyHandler struct with a standard http handler signature `func(rw http.ResponseWriter, req *http.Request)`

The advantage of bringing in the pluginCtx here (see above comment) would be that there would be no need for the RouteInjector step. The DimensionKeys struct can still hold the clientFactory if you want.

You could then use the handler directly in the router, for example `mux.Handle("/dimension-keys", dimensionKeyHandler)` if you name the handler method ServeHTTP.

Since we also discussed where the handler could be created, I will mention here that it could be at or before the router: `mux.Handle("/dimension-keys", routes.NewDimensionKeyHandler(e.getClients))` for example.

Reply:
Same answer as above.